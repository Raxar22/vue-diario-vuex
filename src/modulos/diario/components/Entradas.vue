
<template>
    <div  class="entry-container mb-3 pointer p-2"   
         @click="$router.push({name:'Entrada', params: {id: entrada.id}})">
        <div class="entry-title d-flex">
            <span class="text-success fs-5 fw-bold">{{dia}}</span>
            <span class="mx-1 fs-5">{{mes}}</span>
            <span class="mx-2 fw-ligth">{{anno}}</span>

        </div>
        <div class="entry-description">
            {{textoCorto}}
        </div>
    </div>
</template>

<script>
const meses = ['Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo', 'Junio','Julio', 'Agosto', 'Septiembre', 'Octubre', 'Noviembre', 'Diciembre']
const diasSemana   = ['Domingo','Lunes','Martes','Miércoles','Jueves','Viernes','Sábado']

export default {
    props: {
        entrada: {
            type: Object,
            required: true
        }
    },
    computed:{
        textoCorto(){
            return (this.entrada.text.length > 130)
                ? this.entrada.text.substring(0,130) + '...'
                : this.entrada.text

        },
        dia(){
            const fecha = new Date (this.entrada.fecha)
            return  `${diasSemana[fecha.getDay()]} - ${fecha.getDate()}` 
        },
        mes(){
            const fecha = new Date (this.entrada.fecha)
            return meses[fecha.getMonth()]
        },
        anno(){
            const fecha = new Date (this.entrada.fecha)
            return fecha.getFullYear()
        }
    }
}
</script>

<style lang="scss" scoped>
    .entry-container {
        border-bottom: 1px solid #2c3e50;
        transition: 0.2s all ease-in;
        &:hover{
            background-color: lighten($color: grey, $amount:35 );
            transition: 0.2s all ease-in;
        }
    }
    .entry-description {
        font-size: 12px;
    }

</style>