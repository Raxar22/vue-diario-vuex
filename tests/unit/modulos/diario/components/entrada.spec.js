import { shallowMount } from "@vue/test-utils";
import Entradas from '@/modulos/diario/components/Entradas.vue'
import { diarioState } from "../../../Mocks/test-diario-state";

describe('Pruebas en el componente de Entrada',()=>{

    const mockRouter = {
        push: jest.fn()
    }

    const wrapper = shallowMount (Entradas, {
        props:{
            entrada: diarioState.entries[0]
        },
        global:{
            mocks:{
                $router: mockRouter
            }
        }
    })

    test('Debe hacer match con el snapshot',()=>{
        expect (wrapper.html()).toMatchSnapshot()
    })

    test('Debe de redireccionar al hacer click en el entry-container',()=>{
        const entryContainer = wrapper.find('.entry-container')

        entryContainer.trigger('click')
        expect(mockRouter.push).toHaveBeenCalledWith(
            {
                "name": "Entrada", 
                "params": {
                    "id": diarioState.entries[0].id
                }
            }
        )

    })

    test ('Pruebas en las propiedades computadas',()=>{
        expect(wrapper.vm.dia).toBe('Lunes - 8')
        expect(wrapper.vm.mes).toBe('Agosto')
        expect(wrapper.vm.anno).toBe(2022)
    })
})


