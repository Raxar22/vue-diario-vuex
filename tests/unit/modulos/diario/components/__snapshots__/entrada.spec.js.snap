// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente de Entrada Debe hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">Lunes - 8</span><span class="mx-1 fs-5">Agosto</span><span class="mx-2 fw-ligth">2022</span></div>
  <div class="entry-description">Hola mundo desde mock data</div>
</div>
`;
